### PR TITLE
[feature] Setup API auto doc

### PR DESCRIPTION
**Pull request by zhjwy9343**
*Issue #, if available:*

*Description of changes:*
This PR sets up the new documentation generation mechanism, create new API doc rst files, and modify existing Python code for doc files.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by zhjwy9343** [comment]
the build content pre-view could be found at https://james4graphstorm.readthedocs.io/en/latest/

**Review by thvasilo — COMMENTED** [review]
Getting lots of `WARNING: missing attribute` when building the docs, any idea why those happen?

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+sphinx==7.1.2
+sphinx-rtd-theme==1.3.0
+--extra-index-url https://download.pytorch.org/whl/cpu
+torch==1.13.1+cpu
+-f https://data.dgl.ai/wheels-internal/repo.html
+dgl==1.0.4
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/requirements.txt:6]: In the .readthedocs.yaml file, there is a section to specify to do `pip install .` and this `requirements.txt` is to help readthedocs.io to create a container to build html files from those rst files.

**Comment by zhjwy9343** [comment]
> Getting lots of `WARNING: missing attribute` when building the docs, any idea why those happen?

Oh, I use a template.rst file to let sphinx only read doc string from the given list of methods. Some classes may only have a subset of these methods. So there will be a warning for a missing method. 

This is not an issue.

**Review by thvasilo — APPROVED** [review]
Changes LGTM after clarifications, I built the docs myself and checked out the live preview website.

**Review by jalencato — APPROVED** [review]
LGTM

File: docs/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+sphinx==7.1.2
+sphinx-rtd-theme==1.3.0
+--extra-index-url https://download.pytorch.org/whl/cpu
+torch==1.13.1+cpu
+-f https://data.dgl.ai/wheels-internal/repo.html
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/requirements.txt:5]: These two are for doc building in readthedocs.io only.

**Pull request merged**